1-org/modules/centralized-logging/main.tf

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  key_first_resource         = keys(var.resources)[0]
+  logbucket_sink_member      = { for k, v in var.resources : k => v if k != var.logging_project_key }
+  resource_name              = var.logging_target_type == "bigquery" ? module.destination_bigquery[0].resource_name : var.logging_target_type == "pubsub" ? module.destination_pubsub[0].resource_name : var.logging_target_type == "storage" ? module.destination_storage[0].resource_name : module.destination_logbucket[0].resource_name
+  destination_uri            = length(var.logging_destination_uri) > 0 ? var.logging_destination_uri : var.logging_target_type == "bigquery" ? module.destination_bigquery[0].destination_uri : var.logging_target_type == "pubsub" ? module.destination_pubsub[0].destination_uri : var.logging_target_type == "storage" ? module.destination_storage[0].destination_uri : module.destination_logbucket[0].destination_uri
+  create_destination         = !(length(var.logging_destination_uri) > 0)
+  logging_sink_name          = length(var.logging_sink_name) > 0 ? var.logging_sink_name : "sk-to-${local.logging_target_name_prefix}-${var.logging_destination_project_id}"
+  logging_target_name_prefix = var.logging_target_type == "bigquery" ? "ds" : var.logging_target_type == "pubsub" ? "topic" : var.logging_target_type == "storage" ? "bkt" : "logbkt"
+  logging_target_name        = length(var.logging_target_name) > 0 ? var.logging_target_name : "${local.logging_target_name_prefix}-${random_string.suffix.result}"
+  log_exports                = setunion(local.log_exports_others)
+  parent_resource_ids        = [for parent_resource_id in local.log_exports[*].parent_resource_id : parent_resource_id]
+
+  # Bigquery sink options
+  bigquery_options = var.logging_target_type == "bigquery" && var.bigquery_options != null ? var.bigquery_options : null
+
+  log_exports_others = toset([
+    for value in module.log_export : value
+  ])
+}
+
+resource "random_string" "suffix" {
```

Comment:
why are you adding another random suffix? 
do this new additional 9 characters fit in the max length restrictions of the resources?

Reply:
The module needs this random to generate a logging target name when the user does not provide `var.logging_target_name`.
The name that is generated is: destination-type-prefix + random-string. i.e: ds_audit_logs

Reply:
you mean `ds-qwertyui` ?

big query dataset names do not need a random suffix, they are not global resources, but it is not a problem to have one.

you could use a different default for each type

- "ds_audit_logs_{random_string.suffix.result}"  => can use `_` but not `-`
- "bkt-${module.org_audit_logs.project_id}-audit-logs-${random_string.suffix.result}" => needs more things to prevent collision
- "tp-audit-logs-${random_string.suffix.result}"

Reply:
Done